### ISSUE TITLE
AccessibleSlider: consider adding asserts for step options and roundValue

### ISSUE THREAD

**Issue by mbarlow12** [phetsims/sun#410]
I pulled the rounding out of our check for the keycode. This was causing an issue where we weren't rounding for pgup/pgdown. In GFL, where no `pageKeyboardStep` wasn't explicitly set, that step was being set programmatically and resulting in similar issues that we saw in JT (slider would get stuck from odd rounding issue). Should we consider adding asserts to inform the client that if `roundToStepSize` is active then all keyboard steps should be outlined?

**Comment by jessegreenberg** [comment]
Thanks for the fix, it makes a lot of sense. Could you please clarify what you are recommending for the assertion though? If the roundToStepSize is used, then each of the keyboardStep, shiftKeyboardStep and pageKeyboardStep need to also be defined?

**Comment by mbarlow12** [comment]
@jessegreenberg correct. It's not all that significant, but it might help developers avoid some easy pitfalls.

**Comment by jessegreenberg** [comment]
Understood, thanks, yes making it more strict sounds reasonable, especially since in https://github.com/phetsims/gravity-force-lab-basics/issues/72 we found that roundToStepSize will be used infrequently. And when used, they should be defined to avoid the cases where the value gets "stuck" due to rounding back down to the step size. @mbarlow12 would you mind adding the assertion?

**Comment by jessegreenberg** [comment]
Hmm, I just noticed that the change broke using of the page up/page down keys in RIAW.

**Comment by jessegreenberg** [comment]
Thinking about this more, I don't think we want to round to the nearest step size on larger jumps. In RIAW, that results in rounding to the nearest 1.499, which has odd behavior. If it is important for rounding to occur on larger steps I think we will need separate options to enable/disable that. I am reverting the commit to fix RIAW, and added documentation about rounding only on normal presses. I also checked GFL and didn't noticed any problems with the masses or sliders.

@mbarlow12 if we add assertions like you recommend, would that cover everything without moving the check?
```js
if ( this.roundToStepSize ) {
  newValue = roundValue( newValue, this._valueProperty.get(), stepSize );
}
```

**Comment by mbarlow12** [comment]
What I noticed about the current location of the `roundToStepSize` check is that for pgup/dwn, there's no rounding at all. The mass sliders in GFL will round the default increment and not the pgup/dwn step size, so there's the scenario where shift step moves by 1, default sets to multiple of 50, and page adds/subtracts 100 with no rounding.  If that's ok behavior, then I think assertions might only be necessary to ensure that the default step size is set by the client.

**Comment by jessegreenberg** [comment]
That sounds good @mbarlow12, Ill go ahead and add that assertion. The behavior you described is what is necessary in RIAW/OL, but I don't know if founding to page up/page down will be needed in the future. If so, maybe we can add roundToPageSize (or something) to enable/disable that in the future.

**Comment by jessegreenberg** [comment]
I added the assertion and it is catching lots of usages, good news! It looks like this is because roundtoStepSize was added for all sliders, and I forgot to remove it in https://github.com/phetsims/gravity-force-lab-basics/issues/72.

**Comment by jessegreenberg** [comment]
This was a great idea @mbarlow12, I added the assertion in the above commit. Can you please review, and close if all looks OK?

**Comment by jessegreenberg** [comment]
I should also mention that I let aqua run through ~30 sims before the commit.

**Comment by mbarlow12** [comment]
Sure thing!

**Comment by ariel-phet** [comment]
@jessegreenberg please have someone else review or close

**Comment by jessegreenberg** [comment]
This can be closed.
